File: Tests/TestCases/Events.cs
using System;

public static class Program {
    public static void Main (string[] args) {
        var instance = new CustomType();
        instance.FireEvent(1);
        instance.Event += PrintNumber;
        instance.FireEvent(2);
        instance.Event += PrintNumberTimes2;
        instance.FireEvent(3);
        instance.Event -= PrintNumber;
        instance.FireEvent(4);
        instance.Event -= PrintNumberTimes2;
        instance.FireEvent(5);
    }

    public static void PrintNumber (int x) {
        Console.WriteLine("Event({0})", x);
    }

    public static void PrintNumberTimes2 (int x) {
        Console.WriteLine("Event({0} / 2)", x * 2);
    }
}

public class CustomType {
    public event Action<int> Event;

    public void FireEvent (int x) {
        if (Event != null)
            Event(x);
    }
}

File: Tests/TestCases/Events.vb
Imports System

Public Module Program
    Public Sub Main(ByVal Args As String())
        Dim instance As CustomType = New CustomType()
        instance.FireEvent(1)
        AddHandler instance.Evt, AddressOf PrintNumber
        instance.FireEvent(2)
        AddHandler instance.Evt, AddressOf PrintNumberTimes2
        instance.FireEvent(3)
        RemoveHandler instance.Evt, AddressOf PrintNumber
        instance.FireEvent(4)
        RemoveHandler instance.Evt, AddressOf PrintNumberTimes2
        instance.FireEvent(5)
    End Sub

    Public Sub PrintNumber(ByVal x As Integer)
        Console.WriteLine("Event({0})", x)
    End Sub

    Public Sub PrintNumberTimes2(ByVal x As Integer)
        Console.WriteLine("Event({0} / 2)", x * 2)
    End Sub
End Module

Public Class CustomType
    Public Event Evt As Action(Of Integer)

    Public Sub FireEvent(ByVal x As Integer)
        RaiseEvent Evt(x)
    End Sub
End Class